### PR TITLE
Feature/hw 53097 core create transfer

### PR DESCRIPTION
**Pull request by oleksii-lubianyi-epam**

### CODE REVIEW

File: Tests/Transfer/HyperwalletTransfer.swift [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        XCTAssertEqual(error.getHttpCode(), 400)
+        XCTAssertEqual(error.getHyperwalletErrors()?.errorList?.first?.code, "INVALID_DESTINATION_TOKEN")
+    }
+}
+
+private extension HyperwalletTransferTests {
+    func verifyCreateTransferResponse(_ response: HyperwalletTransfer?) {
+        if let response = response {
+            //Mandatory fields
+            XCTAssertEqual(response.clientTransferId, "6712348070812")
```
Comment (jsaini-hw) [inline Tests/Transfer/HyperwalletTransfer.swift]: Add default messages in the assert statements. like 
`XCTAssertEqual(bankAccountList?.count, 204, "The `count` should be 204")`

File: Sources/Model/Transfer/HyperwalletTransfer.swift [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        case returned               = "RETURNED"
+        case scheduled              = "SCHEDULED"
+        case verificationRequired   = "VERIFICATION_REQUIRED"
+    }
+
+    public let clientTransferId: String
+
+    public let createdOn: String?
+
+    public let destinationAmount: String
```
[the remark this answers is not in the document]
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransfer.swift]: Gotcha, fixed

File: Sources/Model/Transfer/HyperwalletTransfer.swift [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Foundation
+
+public struct HyperwalletTransfer: Codable {
+    public struct HyperwalletForeignExchange: Codable {
```
[the remark this answers is not in the document]
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransfer.swift]: Done

File: Sources/Model/Transfer/HyperwalletTransfer.swift [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        case failed                 = "FAILED"
+        case inProgress             = "IN_PROGRESS"
+        case quoted                 = "QUOTED"
+        case returned               = "RETURNED"
+        case scheduled              = "SCHEDULED"
+        case verificationRequired   = "VERIFICATION_REQUIRED"
+    }
+
+    public let clientTransferId: String
+
```
[the remark this answers is not in the document]
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransfer.swift]: Removed

File: Sources/Model/Transfer/HyperwalletTransfer.swift [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+public struct HyperwalletTransfer: Codable {
+    public struct HyperwalletForeignExchange: Codable {
+        let destinationAmount: String
+        let destinationCurrency: String
+        let rate: String
+        let sourceAmount: String
+        let sourceCurrency: String
+    }
+
+    public enum HyperwalletTransferStatus: String, Codable {
```
[the remark this answers is not in the document]
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransfer.swift]: Moved

File: Sources/Model/Transfer/HyperwalletTransfer.swift [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        case quoted                 = "QUOTED"
+        case returned               = "RETURNED"
+        case scheduled              = "SCHEDULED"
+        case verificationRequired   = "VERIFICATION_REQUIRED"
+    }
+
+    public let clientTransferId: String
+
+    public let createdOn: String?
+
```
[the remark this answers is not in the document]
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransfer.swift]: Removed

File: Sources/Model/Transfer/HyperwalletTransfer.swift [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    public let destinationAmount: String
+
```
[the remark this answers is not in the document]
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransfer.swift]: Removed

File: Sources/Model/Transfer/HyperwalletTransfer.swift [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    public let destinationCurrency: String
+
```
[the remark this answers is not in the document]
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransfer.swift]: Removed

File: Sources/Model/Transfer/HyperwalletTransfer.swift [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+    public let destinationToken: String
+
```
[the remark this answers is not in the document]
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransfer.swift]: Removed

File: Sources/Model/Transfer/HyperwalletTransfer.swift [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+    public let expiresOn: String?
+
```
[the remark this answers is not in the document]
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransfer.swift]: Removed

File: Sources/Model/Transfer/HyperwalletTransfer.swift [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+    public let foreignExchanges: [HyperwalletForeignExchange]?
+
```
[the remark this answers is not in the document]
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransfer.swift]: Removed

File: Sources/Model/Transfer/HyperwalletTransfer.swift [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+    public let memo: String?
+
```
[the remark this answers is not in the document]
Comment (oleksii-lubianyi-epam) [reply Sources/Model/Transfer/HyperwalletTransfer.swift]: Removed

File: Tests/Transfer/HyperwalletTransfer.swift (hunk 1 above)
Comment (oleksii-lubianyi-epam) [reply Tests/Transfer/HyperwalletTransfer.swift]: Added

File: Tests/Transfer/HyperwalletTransfer.swift [hunk 13]
```diff
@@ -0,0 +1,118 @@
+import Hippolyte
+@testable import HyperwalletSDK
+import XCTest
+
+class HyperwalletTransferTests: XCTestCase {
```
Comment (syyang-hw) [inline Tests/Transfer/HyperwalletTransfer.swift]: Can you please rename the file to be HyperwalletTransferTests as well?

Comment (oleksii-lubianyi-epam) [reply Tests/Transfer/HyperwalletTransfer.swift]: Done

**Comment by oleksii-lubianyi-epam** [comment]
Created branch with proper name. New PR to be created

**Pull request closed without merging**